run_prestage.py

```diff
@@ -148,132 +151,105 @@ def get_Xcache_filelist(fnames: list):
-                    # new_sample_list.append("dy_VBF_filter_customJMEoff")
-                    # new_sample_list.append("dy_M-50To120")
-                    # new_sample_list.append("dy_M-120To200")
-                    # new_sample_list.append("dy_VBF_filter_Amandeep10_6_26")
-                    # new_sample_list.append("dy_VBF_filter_Amandeep10_6_32")
-                    # new_sample_list.append("dy_VBF_filter_Amandeep_fromGenSim")
-                    # new_sample_list.append("dy_VBF_filter_fromGridpack")
-                    
-                    # Run3 specific datasets:
-                    # new_sample_list.append("dy_M-50To120")
-                    # new_sample_list.append("dy_M-120To200")
-                
+            for bkg_letter in bkg_samples:
+                for bkg_name in bkg_l:
+                    if bkg_letter in bkg_name:
+                        new_sample_list.append(bkg_name)
+
+        logger.debug(f"new sample list: {new_sample_list}")
 
-                
-                elif bkg_sample.upper() == "TT": # enforce upper case to prevent confusion
-                    new_sample_list.append("ttjets_dl")
-                    new_sample_list.append("ttjets_sl")
-                elif bkg_sample.upper() == "ST": # enforce upper case to prevent confusion
-                    new_sample_list.append("st_tw_top")
-                    new_sample_list.append("st_tw_antitop")
-                elif bkg_sample.upper() == "VV": # enforce upper case to prevent confusion
-                    new_sample_list.append("ww_2l2nu")
-                    new_sample_list.append("wz_3lnu")
-                    new_sample_list.append("wz_2l2q")
-                    new_sample_list.append("wz_1l1nu2q")
-                    new_sample_list.append("zz")
-                elif bkg_sample.upper() == "EWK": # enforce upper case to prevent confusion
-                    new_sample_list.append("ewk_lljj_mll50_mjj120")
-                    # new_sample_list.append("ewk_lljj_mll105_160_ptj0")
-                    # new_sample_list.append("ewk_lljj_mll105_160_py_dipole")
-                else:
-                    print(f"unknown background {bkg_sample} was given!")
-            
         # take sig
         sig_samples = args.sig_samples
+        logger.info(f"signal samples: {sig_samples}")
         if len(sig_samples) >0:
-            for sig_sample in sig_samples:
+            for sig_sample in sig_samples: # FIXME: Why custom? We should just read from YAML file. No hardcoding needed here.
                 if sig_sample.upper() == "GGH": # enforce upper case to prevent confusion
                     new_sample_list.append("ggh_powhegPS")
-                    # new_sample_list.append("ggh_amcPS")
                 elif sig_sample.upper() == "VBF": # enforce upper case to prevent confusion
                     new_sample_list.append("vbf_powheg_dipole")
                 else:
                     print(f"unknown signal {sig_sample} was given!")
```

Comment:
@green-cabbage Same comment here. 

We should just pick information from the YAML file. No need to customize it here.

Reply:
In VBF category DNN training, we use both ggh powheg and ggh amc@NLO samples, so those samples need to exist. I recognize keeping one commented out is not good practice, and make it so that for example, ggh_amcPS isn't available in yaml file, then skip that sample. However, simply not having ggh_amcPS term is not a good idea